locidex/merge.py

```diff
@@ -167,14 +187,73 @@ def write_gene_fastas(seq_data,work_dir):
             oh.write(f'>{seq_name}\n{seq}\n')
             oh.close()
             d+=1
-    return files   
+    return files
+
+def read_samplesheet(sample_key_file):
+    """
+    We will be accepting a file with two columns either ['sample', 'mlst_alleles'] or ['sample_name', 'mlst_alleles']
+    """
+    try:
+        df = pd.read_csv(sample_key_file, sep=",", usecols= lambda column: column in {"sample", "sample_name", "mlst_alleles"}, header=0)
+        df["mlst_alleles"] = df["mlst_alleles"].apply(lambda x: os.path.basename(x))
+        sampledict = df.set_index("mlst_alleles").T.to_dict('list')
+
+
+    except:
+            logging.critical("File should be CSV that will contain either ['sample', 'mlst_alleles'] or ['sample_name', 'mlst_alleles']")
+            raise FileNotFoundError("Incorrect file format")
+
+    return sampledict
+
+def validate_profiles(mlst, sample_id, file_name):
```

Comment:
I am wondering if this could be changed to something like `validate_and_fix_profiles`?

My reasoning is that the passed `mlst` object will get modified by this function. Normally something named `validate` I would not expect to modify the inputs. But something with `fix` in the name I would. So the name change just makes this clearer that the input `mlst` gets modified.